waspc/README.md

```diff
@@ -17,150 +17,135 @@ If you would like to make your first contribution, here is a handy checklist we
-to build Wasp with.
+### Repo
 
-```cabal
-with-compiler: ghc-x.y.z
-```
+Clone this repo (or its fork) to your machine.
 
-Please ensure that you have this ghc version installed.
+Position yourself in this directory (`waspc/`) and make sure that you are on the `main` branch.
 
-### Repo
+### `run` script
 
-Fork this repo and clone the fork to your machine (or clone this repo directly if you don't plan to contribute but just want to try it out).
+[./run](run) script captures the most common commands / workflows for developoment of `waspc` (similar to the role that `scripts` in `package.json` play in `npm` projects), e.g. `./run build`, `./run test`, ... .
 
-Position yourself in this directory (`waspc/`) and make sure that you are on the `main` branch.
+It serves both as a convenient way to easily run common development tasks and also as a sort of a documentation on how we develop `waspc`. You will see us mentioning it often in the rest of this README.
 
-### Build
+Running `./run` without any arguments will print help/usage, which is a good way to discover the typical `waspc` development workflows.
 
-```
-cabal build
-```
+> [!TIP]
+> To make it easy to run the `./run` script from any place in your waspc codebase, you can create a bash alias that points to it, e.g.:
+> ```sh
+> alias wrun="/home/martin/git/wasp-lang/wasp/waspc/run"
+> ```
 
-to build the library and `wasp` executable.
+### Setup
 
-This might take a while (10 mins) if you are doing it for the very first time, since `cabal` will need to download the external dependencies.
-If that is the case, relax and feel free to get yourself a cup of coffee! When somebody asks what you are doing, you can finally rightfully say "compiling!" :D.
+#### Haskell toolchain
 
-:warning: You may need to run `cabal update` before attempting to build if it has been some time since your last update.
+The best way to install Haskell toolchain is via [ghcup](https://www.haskell.org/ghcup/).
 
-:warning: If you are on Mac and get "Couldn't figure out LLVM version!" error message while building, make sure you have LLVM installed and that it is correctly exposed via env vars (PATH, LDFLAGS, CPPFLAGS). The easiest way to do it is by just running `brew install llvm@13`, this should install LLVM and also set up env vars.
+Once you have `ghcup` installed, run `ghcup tui` command to install and select correct versions of `cabal` (package manager), `hls` (language server) and `ghc` (compiler).
 
-:warning: If the LLVM error persists even after its installation, you may need to manually add it your PATH. To do this, you should add the following to end of your shell rc file (e.g. _~/.bashrc_ or _~/.zshrc_): `export PATH="/opt/homebrew/Cellar/llvm@13/13.0.1_2/bin/:$PATH"`.
+To find out what the correct versions of these tools are for the `waspc`:
+ - Run `./run ghcup-set` to check which exact versions of `ghc` and `hls` you should use.
+ - As for `cabal`, just go with the latest one.
```

Comment:
This is new thing I added, how to figure out versions, since I figured out I actually don't know how to figure them out hah.